### ISSUE TITLE
Current approach to multi-file programs is broken

### ISSUE THREAD

**Issue by dvanhorn** [dvanhorn/crook#2]
Multi-file programs are currently broken because if you have a module M1 with variants `A` and it requires module M2 which has variants `A,B`, you will only get a variant `A` of M1, but really you want `A` and `B`.

Likewise if you have M1 with `A,B,C` and M2 with `A` and M3 with `A,B,C`, you will get an `A,B,C` variant of M1, but it will always use variant `A` of M2, which in turn uses only `A` of M3.  You really want M2 to have `A,B,C` variants, which each require the corresponding variant of M3.

In other words, every module should have a variant for every variant it depends upon, plus any addition ones it adds.  For each additional variation, it should use the greatest variation that is less than the new one.

So in the M1, M2, M3 example, M2 would have `A,B,C` because M3 has `A,B,C`.  If we were to add a `B1` to M1, there would be no additional M2 or M3 variants; the M1 `B1` variant would use the `B` variant of M2 (and M3).

**Comment by dvanhorn** [comment]
This was solved by requiring an explicit list of variants.  A little tedious, but very easy.
